connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorMaker.java

```diff
@@ -322,6 +324,29 @@ public void run() {
         }
     }
 
+    /**
+     * Validates that the underlying connectors and tasks are running.
+     * Visible for testing purpose.
+     */
+    boolean isConfiguredAndRunning() {
+        return CONNECTOR_CLASSES.stream().allMatch(connectorClazz -> {
+            final String connName = connectorClazz.getSimpleName();
+            return herderPairs.stream().allMatch(sourceAndTarget -> {
+                final ConnectorStateInfo connectorStatus = this.connectorStatus(sourceAndTarget, connName);
+                return connectorStatus != null
+                    // verify that connector state is set to running
+                    && connectorStatus.connector().state().equals(AbstractStatus.State.RUNNING.toString())
+                    // verify that all tasks are set to running
+                    && connectorStatus.tasks().stream().allMatch(s -> s.state().equals(AbstractStatus.State.RUNNING.toString()));
+            });
+        });
+    }
+
+    private ConnectorStateInfo connectorStatus(SourceAndTarget sourceAndTarget, String connector) {
+        checkHerder(sourceAndTarget);
+        return herders.get(sourceAndTarget).connectorStatus(connector);
+    }
```

Comment:
This can be made `public` and we can move all other assertion logic around the status of MM2 into testing code.